### ISSUE TITLE
Implement cylindrical PML

### ISSUE THREAD

**Issue by RemiLehe** [ECP-WarpX/WarpX#904]
The PML are currently not implemented in RZ geometry (this is true for both the FDTD and PSATD modes).

The FDTD implementation could be done by following a similar approach as in [this paper](https://www.sciencedirect.com/science/article/pii/S002199919896082X). Note that the above paper studies only the case `m=0` ; hence the theory would have to be extended to azimuthal mode decomposition.
Note also that the above-mentioned implementation should be done after #808 (which refactors the FDTD implementation of the PML - in Cartesian geometry).

The PSATD implementation could be done by following [the same approach as FBPIC](https://github.com/fbpic/fbpic/pull/417)

**Comment by dpgrote** [comment]
@RemiLehe Is there a paper describing the method used in FBPIC? I can do this, after implementing the Galilean and other updates in the queue for the RZ spectra.

**Comment by RemiLehe** [comment]
No, the FBPIC implementation is new and unpublished so far. However, I can send some notes by email.

**Comment by RemiLehe** [comment]
The PSATD version of the PML has been implemented in https://github.com/ECP-WarpX/WarpX/pull/2211.

**Comment by RemiLehe** [comment]
Another paper relevant for the RZ FDTD implementation:
https://www.sciencedirect.com/science/article/abs/pii/S0010465525002395
